forth.asm

```diff
@@ -203,6 +203,28 @@ dict_entry tonumber, '>NUMBER', 0x0 ;( str -- n)
     push    ebx
     jmp     next
 
+interpret_if:
+    ; if stack nozero, advance instruction pointer
+    pop     eax
+    test    eax, eax
+    jnz     .true
+    add     esi, 0x8
+    jmp     next
+.true:
+    jmp     next
+
+
+dict_entry interpret, 'INTERPRET-WORD', 0x0 ; ( str -- )
+    call    docolon
+    dd      blank
+    dd      cword
+    dd      find
+    dd      interpret_if
```

Comment:
That is what I did already, see [above](https://github.com/zormit/smorth/pull/1/files/c8e95ef14d9536e3a1ae4fd2ef8da69e5d29feac..9d10ffa088b5139ed3615b5ab2064f7f3884b139#diff-4f1b15a588a349787acbd2e56d41d6d1R206). Seems to be the right way then... Modifying the Forth PC felt too hacky to me. But that's what I am always saying... Forth is so hacky... How does it even work properly? :P